src/Product/WCProductAdapter.php

```diff
@@ -903,4 +912,166 @@ public function setTargetCountry( $targetCountry ) {
 		// product shipping information is also country based
 		$this->map_wc_product_shipping();
 	}
+
+	/**
+	 * Performs the attribute mapping.
+	 * This function applies rules setting values for the different attributes in the product.
+	 *
+	 * @param array $mapping_rules The set of rules to apply
+	 */
+	protected function map_attribute_mapping_rules( array $mapping_rules ) {
+		$attributes = [];
+
+		if ( empty( $mapping_rules ) ) {
+			return $this;
+		}
+
+		$this->set_product_category_ids();
+
+		foreach ( $mapping_rules as $mapping_rule ) {
+			if ( $this->rule_match_conditions( $mapping_rule ) ) {
+				$attribute_id                = $mapping_rule['attribute'];
+				$attributes[ $attribute_id ] = $this->format_attribute(
+					apply_filters(
+						"woocommerce_gla_product_attribute_value_{$attribute_id}",
+						$this->get_source( $mapping_rule['source'] ),
+						$this->get_wc_product()
+					),
+					$attribute_id
+				);
+			}
+		}
+
+		parent::mapTypes( $attributes );
+
+		// Size
+		if ( ! empty( $attributes['size'] ) ) {
+			$this->setSizes( [ $attributes['size'] ] );
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Set the product category ids property
+	 */
+	protected function set_product_category_ids() {
+		$terms                      = get_the_terms( $this->get_wc_product()->get_id(), 'product_cat' );
+		$this->product_category_ids = $terms ? wp_list_pluck( $terms, 'term_id' ) : [];
+	}
+
+	/**
+	 * Get a source value for attribute mapping
+	 *
+	 * @param string $source The source to get the value
+	 * @return string|null The source value for this product
+	 */
+	protected function get_source( string $source ) {
+		$source_type = null;
+
+		$type_separator = strpos( $source, ':' );
+
+		if ( $type_separator ) {
+			$source_type  = substr( $source, 0, $type_separator );
+			$source_value = substr( $source, $type_separator + 1 );
+		}
+
+		switch ( $source_type ) {
+			case 'product':
+				return $this->get_product_field( $source_value );
+			case 'taxonomy':
+				return $this->get_product_taxonomy( $source_value );
+			case 'attribute':
+				return $this->get_wc_product()->get_meta( $source_value );
+			default:
+				return $source;
```

Comment:
At this point, in case the source type is not in product|taxonomy|attribute we take the source as static value. 

The custom attributes like yoast: should be treated independently as you mentioned

Reply:
Btw. I added the comment